ReText/tab.py

```diff
@@ -444,15 +444,76 @@ def openSourceFile(self, fileToOpen):
 		"""Finds and opens the source file for link target fileToOpen.
 
 		When links like [test](test) are clicked, the file test.md is opened.
+		# TODO: broken objective? See TODO below.
 		It has to be located next to the current opened file.
 		Relative paths like [test](../test) or [test](folder/test) are also possible.
+
+		:param fileToOpen:	Link target
+							This can be a file of any extension or a path
+		:returns: 			Filename successfully handled to openFileWrapper
+							or 'noSuchFile' to stop further processing.
+		:rtype: 			str
 		"""
 		if self.fileName:
+			# Please give description of this condition.
+			# (e.g. "if internal link was clicked from within valid open document in a tab"?)
 			currentExt = splitext(self.fileName)[1]
 			basename, ext = splitext(fileToOpen)
 			if ext in ('.html', '') and exists(basename + currentExt):
+				# TODO: It seems this should catch markdown links like
+				#  [test](test) but than ext not in ('.html', '')
 				self.p.openFileWrapper(basename + currentExt)
 				return basename + currentExt
-		if exists(fileToOpen) and get_markup_for_file_name(fileToOpen, return_class=True):
+
+		if get_markup_for_file_name(fileToOpen, return_class=True):
+			if not QFile.exists(fileToOpen):
+				# ReText should only support to create markup-files, no plain
+				# text files. Thereby it facilitates writing cross-linked
+				# documentation.
+				# Consequently create-file functionality can be nested here.
+				if not self.createFileQuestion(fileToOpen):
+					# File still not exists
+					return 'noSuchFile'
+
+			# fileToOpen exists and has markup
 			self.p.openFileWrapper(fileToOpen)
 			return fileToOpen
+
+	def createFileQuestion(self, fileToCreate):
+		"""Ask user if file should be created and do so if answer is yes.
+
+		:returns: True if new file has been created, False otherwise
+		:rtype: bool
+		"""
+		buttonReply = QMessageBox.question(self, self.tr('Create missing file?'),
+			self.tr("The file '%s' does not exist.\n\nDo you want to create it?" % fileToCreate),
+			QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
+		if buttonReply == QMessageBox.Yes:
+			if self.createFile(fileToCreate):
+				return True
+			else:
+				return False
+		elif buttonReply == QMessageBox.No:
+			return False
+
+	def createFile(self, fileToCreate):
+		"""Try to create file, return True if successful"""
+		try:
+			# Create file:
+			open(fileToCreate, 'x').close()
+			return True
+		except FileNotFoundError:
+			# Most expected error due possible misspelling in path and
+			# should thus be handled separately.
+			# 'No such file or directory' reduces to 'No such directory'
+			# because we call createFile if not exists(fileToOpen)
+			QMessageBox.warning(self, 'No such directory',
+								self.tr("File could not be created: '%s'" % fileToCreate))
+			return False
+		except IOError as errormsg:
+			# Does ReText support Python <3.3? If not change IOError to OSError.
```

Comment:
We support Python 3.2 and later currently (see README). You can change this in a separate pull request if you want.